MoEngagePluginBase/MoEngagePluginUtils.swift

```diff
@@ -137,10 +127,41 @@ public class MoEngagePluginUtils {
         } else if inappAction.actionType == NavigationAction {
             actionType =  MoEngagePluginConstants.General.navigation
         }
-         
+        
         return actionType
     }
-}
+    
+    static func mapSelfHandledCampaignDataToJSON(campaignData: MoEngageInAppSelfHandledData) -> [String: Any] {
+        let accountMeta = campaignData.accountMeta
+        let accountPaylaod = createAccountPayload(identifier: accountMeta.appID)
+       
+        let campaignsPayload = campaignData.campaigns.map { campaign in
+            let selfHandledPayload = mapSelfHandledCampaignToJSON(campaign)
+            return [MoEngagePluginConstants.General.platform: MoEngagePluginConstants.General.iOS,
+                    MoEngagePluginConstants.General.accountMeta: accountPaylaod,
+                    MoEngagePluginConstants.General.data: selfHandledPayload]
+        }
+        
+        return [MoEngagePluginConstants.General.accountMeta: accountPaylaod, MoEngagePluginConstants.InApp.campaigns: campaignsPayload] as [String : Any]
+    }
+    
+    private static func mapDisplayRulesToJSON(_ rules: MoEngageInAppRules) -> [String: Any] {
+        return [MoEngagePluginConstants.InApp.screenName: rules.screenName, MoEngagePluginConstants.InApp.contexts: rules.contexts]
+    }
+
+    static func mapSelfHandledCampaignToJSON(_ campaign: MoEngageInAppSelfHandledCampaign) -> [String: Any] {
+        var campaignPayload = campaign.fetchInAppPaylaod()
+      
+        var selfHandledPayload = [String: Any]()
+        selfHandledPayload[MoEngagePluginConstants.General.payload] = campaign.campaignContent
+        selfHandledPayload[MoEngagePluginConstants.InApp.dismissInterval] = campaign.autoDismissInterval
+        selfHandledPayload[MoEngagePluginConstants.InApp.displayRules] = mapDisplayRulesToJSON(campaign.displayRules)
```

Comment:
Is it fine to add display rules for normal self handled campaigns  also ?

Reply:
@babulmoengage its added for existing self handled api

Reply:
@babulmoengage its added for existing self handled api. This is common code

Reply:
@babulmoengage its added for existing self handled api. This is common code